Add Github Action workflow for Robottelo

Gauravtalreja1 opened the pull request:
**Proposal for:**

- [x] Adding a Github Action workflow for Robottelo.

- [x] Dropping/Decommissioning Travis-CI from Robottelo.

- [x] Adding a `broker_directory` in `robottelo.properties.sample`.

- [x] Adding a blank `virtwho.properties.sample` file to workaround https://github.com/SatelliteQE/robottelo/issues/8138 .

**Results**
Github Action run will look like:
https://github.com/Gauravtalreja1/robottelo/actions/runs/349472736
https://github.com/Gauravtalreja1/robottelo/pull/1/checks



Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

omkarkhatavkar commented on .github/workflows/action.yml:
```diff
@@ -0,0 +1,61 @@
+
+jobs:
+  codechecks:
+    name: 📃 Code Quality
```
Looks like the icon are getting detected here

mshriver replied:
I'd prefer to not have emotes in our pipeline names.

omkarkhatavkar commented:
@Gauravtalreja1 Let's keep the Travis file for some more time, we can remove later @mshirver thoughts?

mshriver commented:
https://github.com/SatelliteQE/robottelo/pull/8144

You'll see some failures in the test collection until this is merged.

JacobCallahan commented:
The travis config could be renamed instead of straight up removed, just to make it easy to bring it back (just in case).
If things go well for a while, it can be removed completely.

JacobCallahan commented on .github/workflows/action.yml:
```diff
@@ -0,0 +1,61 @@
+jobs:
+  codechecks:
+    name: 📃 Code Quality
+    runs-on: ubuntu-latest
```
Can we change this to a fedora image?

omkarkhatavkar replied:
Looks like they dont support it out of the box, I saw this thread https://github.com/actions/virtual-environments/issues/1443, Any specific reasons we have to use this with fedora image? because we were using xenial Travis which is ubuntu distribution. I don't think we should be blocking this against fedora image.

mshriver replied:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

Not without changing the pipeline to be a docker image strategy. Github offers ubuntu VMs for these pipelines. Same as in travis, we're running on ubuntu.

JacobCallahan commented on .github/workflows/action.yml:
```diff
@@ -0,0 +1,61 @@
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get install -y libgnutls28-dev libcurl4-openssl-dev libssl-dev
```
If we can do fedora, this will need to be changed

omkarkhatavkar replied:
If you want to support fedora we have matrix kind of thing. but I guess not that necessary.

JacobCallahan reviewed (changes requested):
If possible, Fedora is preferred.

Gauravtalreja1 commented:
**New Changes:**
1. As per suggestion from @lpramuk , updated workflow to use `--setup-plan` in test collection rather than simple `--collect-only`.
2. Keeping a Travis config `.travis.yml` for some more time, for some more PR's.
3. Removed all emoticons from pipeline/task names.
3. Discarding changes done in `robottelo.properties.sample` as it was already there in that file.

**Latest Run:** https://github.com/Gauravtalreja1/robottelo/pull/1/checks

omkarkhatavkar reviewed (approved):
ACK

JacobCallahan reviewed (approved):
ACK
